File: src/BaseHmiTypes/Screens/Widgets/HmiClock.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Widgets;

public class HmiClock : HmiWidgetBase
{
    public HmiClock()
    {
        HmiObjectType = BaseHmiTypes.Screens.Base.HmiObjectType.HmiClock;
    }

    public HmiProperty<bool>? Analog { get; set; }

    public HmiProperty<int>? NumberStyle { get; set; }

    public HmiProperty<bool>? ShowDate { get; set; }

    public HmiProperty<bool>? ShowTime { get; set; }

    public HmiProperty<bool>? ShowHours { get; set; }

    public HmiProperty<bool>? ShowMinutes { get; set; }

    public HmiProperty<bool>? ShowSeconds { get; set; }

    /// <summary>
    /// Product-specific date/time display format retained without locale-dependent conversion.
    /// </summary>
    public HmiProperty<string>? Format { get; set; }

    public HmiProperty<string>? TimeZone { get; set; }
}

File: typescript/base-hmi-types/src/screens/widgets/HmiClock.ts
import { HmiProperty } from "../base.js";
import { HmiWidgetBase } from "./HmiWidgetBase.js";
import { HmiObjectType } from "../base/HmiObjectType.js";

export class HmiClock extends HmiWidgetBase {
  constructor() {
    super();
    this.hmiObjectType = HmiObjectType.HmiClock;
  }

  analog?: HmiProperty<boolean>;
  numberStyle?: HmiProperty<number>;
  showDate?: HmiProperty<boolean>;
  showTime?: HmiProperty<boolean>;
  showHours?: HmiProperty<boolean>;
  showMinutes?: HmiProperty<boolean>;
  showSeconds?: HmiProperty<boolean>;
  /** Product-specific date/time format retained without locale-dependent conversion. */
  format?: HmiProperty<string>;
  timeZone?: HmiProperty<string>;
}
